cclib/parser/orcaparser.py

```diff
@@ -1339,9 +1339,61 @@ def energy_intensity(line):
+                #
+                # Don't parse from SOC sections.
+                # ROCIS COMBINED is combination of SOC and ROCIS (we still parse the normal ROCIS section).
+                if not any([soc_header in name for soc_header in ["SPIN ORBIT CORRECTED", "SOC CORRECTED", "ROCIS COMBINED"]]):
+                    # We need to be careful about how we parse etenergies from these spectrum sections.
+                    # First, and in most cases, energies printed here will be the same as those printed in 
+                    # previous sections. The energies in cm-1 aught to match exactly to those parsed previously,
+                    # but other units may have rounding errors.
+                    # Secondly, some methods (ROCIS, CASSCF, SOC to name a few) may only print their final excited state
+                    # energies in this spectrum section, in which case the energies will not match those previously parsed
+                    # (which will be from lower levels of theory that we're not interested in). This means we cannot simply
+                    # ignore the energies printed. Also, in this case we must decide whether to discard other previously
+                    # parsed etdata (etsyms, etsecs etc).
+                    # Thirdly, SOC prints spin-mixed excited state spectra. This is interesting, but does not match the 
+                    # number of states or symmetry of data parsed in previous sections, so is not used to overwrite etenergies.
+                    
+                    # If we have no previously parsed etnergies, there's nothing to worry about.
+                    if not hasattr(self, "etenergies"):
+                        self.set_attribute("etenergies", etenergies)
+                    
+                    # Determine if these energies are same as those previously parsed.
+                    # May want to use a smarter comparison?
+                    elif len(etenergies) == len(self.etenergies) and all([self.etenergies[index] == etenergy for index, etenergy in enumerate(etenergies)]):
+                        pass
+                    
+                    # New energies.
+                    else:
+                        # Because these energies are new, we do not know if they correspond to the same level of theory
+                        # as the previously parsed etsyms etc.
+                        self.logger.warning("New excited state energies encountered in spectrum section, resetting excited state attributes")
+                        
+                        for attr in ("etenergies", "etsyms", "etoscs", "etsecs", "etrotats"):
+                            if hasattr(self, attr):
+                                delattr(self, attr)
+                                
+                        self.set_attribute("etenergies", etenergies)
+                    
+#                     # There's no point overwriting the etenergies we've already parsed,
```

Comment:
I can understand wanting to leave in a comment like this, but can you clean it up a little? Something like maybe, mostly the conditional on 1390/1391 is confusing. I also needed to remind myself that `etenergies` is indeed in scope for the entire block of this PR.
```patch
@@ -1383,13 +1383,12 @@ States  Energy Wavelength    D2        m2        Q2         D2+m2+Q2       D2/TO
                                 
                         self.set_attribute("etenergies", etenergies)
                     
-#                     # There's no point overwriting the etenergies we've already parsed,
-#                     # especially because the previously parsed values normally have greater precision.
-#                     # We could do a check to make sure the energies match, but we will fall foul
-#                     # of rounding errors.
-#                     #if not hasattr(self, "etenergies"):
-#                     if True:
-#                         self.set_attribute("etenergies", etenergies)
+                    # There's no point overwriting the etenergies we've already parsed,
+                    # especially because the previously parsed values normally have greater precision.
+                    # We could do a check to make sure the energies match, but we will fall foul
+                    # of rounding errors.
+                    # if not hasattr(self, "etenergies"):
+                    #     self.set_attribute("etenergies", etenergies)
                     
                     self.set_attribute('etoscs', etoscs)
                 
```

Reply:
Sorry I can actually remove most of this comment (and obviously the test is just from debugging). The energies printed in cm-1 are actually identical in both sections, so less of a problem than I initially thought (:

In terms of scoping, yes I also find this section a bit busy and confusing. In the future I think it would benefit from a re-write/tidy-up, but this can probably wait until after it's decided where all these different excited state energies should live long-term